commit df5af119c5fc91522b7f6eddf1845b6b3b485df8
Author: James Page <james.page@chainguard.dev>
Date:   Fri Nov 28 12:09:58 2025 +0000

    Add jdk.random to BOOT_MODULES for JRE inclusion
    
    Include jdk.random module in the boot modules list to make
    RandomGenerator implementations available in JRE images by default.
    
    This improves backwards compatibility with future JDK versions where
    jdk.random module was removed and implementations moved to java.base
    (JDK-8330005 in JDK 23). Applications using RandomGeneratorFactory
    will now work correctly in minimal JRE images without requiring
    explicit jlink module configuration.
    
    Without this change, RandomGeneratorFactory.getDefault() throws
    NoSuchElementException in java.base-only runtime images.

diff --git a/make/conf/module-loader-map.conf b/make/conf/module-loader-map.conf
index 2c95b8570fd..2a713b63566 100644
--- a/make/conf/module-loader-map.conf
+++ b/make/conf/module-loader-map.conf
@@ -51,6 +51,7 @@ BOOT_MODULES= \
     jdk.management.agent \
     jdk.net \
     jdk.nio.mapmode \
+    jdk.random \
     jdk.sctp \
     jdk.unsupported \
     jdk.naming.rmi \
